File: xamarin/android/PropertyCross/NavigationService.cs
using System;
using Android.App;
using Android.Content;
using PropertyCross.Presenter;
using com.propertycross.xamarin.android.Views;

namespace com.propertycross.xamarin.android
{
	public class NavigationService : INavigationService
	{
		private PropertyCrossApplication application;

		public NavigationService(PropertyCrossApplication app)
		{
			application = app;
		}

		public void PushPresenter(object presenter)
		{
			object oldPresenter = application.Presenter;
			if(presenter != oldPresenter)
			{
				application.Presenter = presenter;
				Intent i = null;

				if(presenter is SearchResultsPresenter)
				{
					i = new Intent(application.CurrentActivity, typeof(SearchResultsView));
				}
				else if(presenter is PropertyPresenter)
				{
					i = new Intent(application.CurrentActivity, typeof(PropertyView));
				}
				else if(presenter is FavouritesPresenter)
				{
					i = new Intent(application.CurrentActivity, typeof(FavouritesView));
				}

				if(i != null)
					application.CurrentActivity.StartActivity(i);
			}
		}
	}
}

File: native/android/PropertyCross/src/com/propertycross/android/services/NavigationService.java
package com.propertycross.android.services;

import android.content.Intent;

import com.propertycross.android.presenter.FavouritesPresenter;
import com.propertycross.android.presenter.INavigationService;
import com.propertycross.android.presenter.PropertyPresenter;
import com.propertycross.android.presenter.SearchResultsPresenter;
import com.propertycross.android.views.FavouritesView;
import com.propertycross.android.views.PropertyFinderApplication;
import com.propertycross.android.views.PropertyView;
import com.propertycross.android.views.SearchResultsView;

public class NavigationService implements INavigationService {
	
	private PropertyFinderApplication application;

	public NavigationService(PropertyFinderApplication app)
	{
		application = app;
	}

	public void pushPresenter(Object presenter)
	{
		Object oldPresenter = application.presenter;
		if(presenter != oldPresenter)
		{
			application.presenter = presenter;
			Intent i = null;

			if(presenter instanceof SearchResultsPresenter)
			{
				i = new Intent(application.currentActivity, SearchResultsView.class);
			}
			else if(presenter instanceof PropertyPresenter)
			{
				i = new Intent(application.currentActivity, PropertyView.class);
			}
			else if(presenter instanceof FavouritesPresenter)
			{
				i = new Intent(application.currentActivity, FavouritesView.class);
			}

			if(i != null)
				application.currentActivity.startActivity(i);
		}
	}
}